tests/TestCase/View/Helper/PaginatorHelperTest.php

```diff
@@ -3381,6 +3381,42 @@ public function testLimitControl()
         $this->assertHtml($expected, $out);
     }
 
+    /**
+     * test the limitControl() with a request url and query.
+     *
+     * @return void
+     */
+    public function testLimitControlUrlWithQuery()
+    {
+        $request = new ServerRequest([
+            'url' => '/batches?owner=billy&expected=1',
+            'params' => [
+                'plugin' => null, 'controller' => 'Batches', 'action' => 'index', 'pass' => [],
+            ],
+            'base' => '',
+            'webroot' => '/',
+        ]);
+        Router::setRequest($request);
+        $this->View->setRequest($request);
+
+        $out = $this->Paginator->limitControl([1 => 1]);
+        $expected = [
+            ['form' => ['method' => 'get', 'accept-charset' => 'utf-8', 'action' => '/batches?owner=billy&amp;expected=1']],
```

Comment:
We would expect `/batches?owner=billy&expected=1`

Reply:
Why? You're in HTML. A bare `&` is not allowed. The [HTML4 spec](https://www.w3.org/TR/html401/charset.html#h-5.3.2) has a section on this. That section is also present in the [HTML 5 specs](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2).

Reply:
I read the standard as saying you can't have a & with identifier-style letters and numbers and then `;`.

Reply:
Yes, but there isn't a reasonable way to ensure there is no `;` somewhere in the trailing values. PHP's built-in HTML escaping behaves in a similar way

```php
php > echo htmlspecialchars('foo&bar', ENT_QUOTES);
foo&amp;bar
php > echo htmlspecialchars('foo&bar', ENT_HTML5);
foo&amp;bar
```
Using alternative HTML escaping presents a number of risks an unknown consequences to applications.